staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -682,9 +684,15 @@ func TestCacherNoLeakWithMultipleWatchers(t *testing.T) {
 			}
 		}
 	}()
+	go func() {
```

Comment:
Sorry I do not follow why we need a new go routine here?

Reply:
this is so we don't block and allow receive on errorCh on line 693